File: DSA/SakshamDwivedi_590013147/Day19/D19P3.c
#include <stdio.h>

int main() {
    int matrix[100][100];
    int rows; 
    int cols; 
    int target;

    printf("Enter number of rows: ");
    scanf("%d", &rows);
    printf("Enter number of columns: ");
    scanf("%d", &cols);

    printf("Enter the array elements (row-wise):");
    for (int i = 0; i < rows; i++) {
        for (int j = 0; j < cols; j++) {
            scanf("%d", &matrix[i][j]);
        }
    }

    printf("Enter the target value to find: ");
    scanf("%d", &target);

    int row = 0;
    int col = cols - 1; 
    int found = 0;

    while (row < rows && col >= 0) {
        if (matrix[row][col] == target) {
            found = 1;
            break;
        } else if (matrix[row][col] > target) {
            col--;
        } else {
            row++;
        }
    }

    printf("Matrix:\n");
    for (int i = 0; i < rows; i++) {
        for (int j = 0; j < cols; j++) {
            printf("%d ", matrix[i][j]);
        }
        printf("\n");
    }

    if (found) {
        printf("True");
    } else {
        printf("False");
    }
}

File: DSA/SakshamDwivedi_590013147/Day19/D19P3.py
#Solution Function
def search_matrix(matrix, target):
    rows = len(matrix)
    cols = len(matrix[0])
    
    row = 0
    col = cols - 1
    while row < rows and col >= 0:
        if matrix[row][col] == target:
            return True
        elif matrix[row][col] > target:
            col -= 1
        else:
            row += 1

    return False

#Fixed Input
matrix = [[1,4,7],[2,5,8],[3,6,9]]
print(search_matrix(matrix, 3))

#Custom Input
rows = int(input("Enter number of rows: "))
cols = int(input("Enter number of columns: "))
arr_pro = list(int(x) for x in input("Enter the array elements: ").split())
arr = [arr_pro[i *cols:(i + 1)*cols] for i in range(rows)]
target = int(input("Enter the target value to find : "))
for row in arr:
    print(row)
print(search_matrix(matrix, target))
